### DISCUSSION TITLE
Bad first layer

### DISCUSSION THREAD

**Discussion by frediz** [xblax/flashforge_ad5m_klipper_mod#214] [Q&A]
Hi,
first off I'm not a 3d printing expert and I'm a klipper newbie.
After installing the mod I tried a few PETG prints and they almost all failed.
The bed was at 70°C and the extruder 255°C.
For one model, after 2-3 layers, the model came off 5 times out of 6.
For the other model, same thing but after 1cm of printed layers .
Any idea on what the reason can be ?
For both, I finally rebooted to the stock firmware and the prints (same gcode) went just fine.
It's just that the [remote] klipper UI is way better...
Thanks,
F.

**Comment by consp** [comment]
I'm no expert either but this was my solution;

Just like in my case the likely source was a too thin first layer. You might need to adjust the offset a tiny bit in printer.cfg:

```
[probe]
z_offset: -0.25
```

See https://www.klipper3d.org/Config_Reference.html#probe

Change it and see how it goes with a proper stick test print. Don't forget do the mesh calibration and load the correct mesh.

YMMV.

I also switched glue so that helped as well, and my PETG sticks like concrete when hot, the PLA also prints fine with a bit more offset and good glue.

**Reply by jeffalanmcclain** [reply]
uhhh....everything I see says that a negative z_offset here actually REDUCES the distance between nozzle and bed, so your example above would reduce it even more (giving more squish), yet you said you had TOO THIN first layer?  Just trying to get this clear.  I've seen vastly different comments on z_offset.  From, "You can't have negative z_offsets" (WtF???) to "When you are talking about a BED raise, negative values actually LOWERS the bed and INCREASES the distance between bed and nozzle (I do NOT believe this is correct and klipper and motor direction all accounts for this on your printer hardware)?

**Reply by consp** [reply]
This is the default, you can change it however you want. It is not a definitive value. You should experiment as every printer is different.

Mine is actually -0.2125. YMMV as always. Try for yourself.
